flutter/lib/src/navigation/sentry_navigator_observer.dart

```diff
@@ -242,14 +274,41 @@ class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
     await _native?.beginNativeFramesCollection();
   }
 
-  Future<void> _finishTransaction() async {
+  Future<void> _startTimeToDisplayTracking(Route<dynamic>? route) async {
```

Comment:
`l`: You could rename this to `_startTransaction` and merge the current content of `_startTransaction` into this function. Now in `didPush` the code reads like this:
```
_finishTransaction();
_startTimeToDisplayTracking(route);
```

Which is a bit confusing because you finish a transaction and you don't start a new one.

Also you could add an early return in here when `!_enableAutoTransactions` what `_startTransaction` already does.

Reply:
I think I'd rather rename `_finishTransaction`, `_startTransaction` is already a rather big function, combining all of them would be a lot imo

Reply:
I don't mind long functions if they clearly communicate what's going on. If the logic is simple and long, I don't mind having it in one function. Up to you.